bash/sanity_checks.bash

```diff
@@ -48,13 +54,58 @@ function Perform_Internal_Sanity_Checks()
     Internally_Ensure_Given_Files_Exist \
         'These Python scripts should be shipped within the hybrid handler codebase.' '--' \
         "${HYBRID_external_python_scripts[@]}"
-    Internally_Ensure_Given_Files_Exist \
-        'These base configuration files should be shipped within the hybrid handler codebase.' '--' \
-        "${HYBRID_software_base_config_file[@]}"
+    for key in "${!HYBRID_software_base_config_file[@]}"; do
+        if [[ "$key" != "Sampler" ]]; then
+            Internally_Ensure_Given_Files_Exist \
+                'These base configuration files should be shipped within the hybrid handler codebase.' '--' \
+                "${HYBRID_software_base_config_file[$key]}"
+        fi
+    done
 }
 
 #===================================================================================================
 
+function __static__Ensure_Valid_Module_Given()
+{
+    if [[ ! "${HYBRID_module[Sampler]}" =~ ^(SMASH|FIST)$ ]]; then
+        exit_code=${HYBRID_fatal_logic_error} Print_Fatal_And_Exit \
+            'The module specified for the Sampler run is not valid.' \
+            'Valid modules are: ' --emph 'SMASH' ' and ' --emph 'FIST' '.'
+    fi
+    return 0
+}
+
+function __static__Ensure_Additional_Paths_Given_For_Sampler()
+{
+    if [[ "${HYBRID_module[Sampler]}" = 'FIST' ]]; then
+        Ensure_Given_Files_Exist \
+            'Some needed file for Thermal-FIST sampler was not specified in the configuration file' \
+            -- "${HYBRID_fist_module[Particle_file]}" "${HYBRID_fist_module[Decays_file]}"
+    fi
+}
+
+function __static__Set_Sampler_Input_Key_Paths()
+{
+    declare -rgA HYBRID_sampler_input_key_default_paths=(
+        # FIST input key paths
+        ['hypersurface_file']="${HYBRID_software_output_directory[Hydro]}/freezeout.dat"
+        ['output_file']="${HYBRID_software_output_directory[Sampler]}/particle_lists.oscar"
+        ['particle_list_file']="${HYBRID_fist_module[Particle_file]}"
+        ['decays_list_file']="${HYBRID_fist_module[Decays_file]}"
+        # SMASH input key paths
+        ['surface']="${HYBRID_software_output_directory[Hydro]}/freezeout.dat"
+        ['spectra_dir']="${HYBRID_software_output_directory[Sampler]}"
+    )
+}
+
+function __static__Choose_Base_Configuration_File_For_Sampler()
+{
+    if [[ "${HYBRID_software_base_config_file[Sampler]}" = '' ]]; then
+        Sampler_key="Sampler_${HYBRID_module[Sampler]}"
+        HYBRID_software_base_config_file[Sampler]="${HYBRID_software_base_config_file[${Sampler_key}]}"
```

Comment:
I guess the variable is for readability. Still, I would comply with names style rules (only lowercase) and make it `local -r`.